src/VisualStudio/Core/Def/Diagnostics/VisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -65,14 +65,15 @@ internal partial class VisualStudioDiagnosticAnalyzerService : IVisualStudioDiag
             _globalOptions = globalOptions;
         }
 
-        public void Initialize(IServiceProvider serviceProvider)
+        public async Task InitializeAsync(IAsyncServiceProvider serviceProvider, CancellationToken cancellationToken)
         {
-            _serviceProvider = serviceProvider;
+            _serviceProvider = (IServiceProvider)serviceProvider;
```

Comment:
not sure if this cast is safe.  The 'servicePRovider' to all these calls it the RoslynPackage instance, so i would hope this cast could be on any thread.  if not, i can pass in both teh sync and async service provdier to all these calls.

Reply:
Roslyn should not be using `IServiceProvider` APIs.